compiler/test/data/typescript/node_modules/interface/constructSignature/singleConstructorWithGenericDefault.d.kt

```diff
@@ -0,0 +1,32 @@
+// [test] singleConstructorWithGenericDefault.module_resolved_name.kt
+@file:Suppress("INTERFACE_WITH_SUPERCLASS", "OVERRIDING_FINAL_MEMBER", "RETURN_TYPE_MISMATCH_ON_OVERRIDE", "CONFLICTING_OVERLOADS")
+
+import kotlin.js.*
+import org.khronos.webgl.*
+import org.w3c.dom.*
+import org.w3c.dom.events.*
+import org.w3c.dom.parsing.*
+import org.w3c.dom.svg.*
+import org.w3c.dom.url.*
+import org.w3c.fetch.*
+import org.w3c.files.*
+import org.w3c.notifications.*
+import org.w3c.performance.*
+import org.w3c.workers.*
+import org.w3c.xhr.*
+
+external interface A<T> {
+    fun foo()
+}
+
+external interface AConstructor {
+    fun bar()
+}
+
+external open class Z<T>(a: T) : A<T> {
+    companion object : AConstructor
+}
+
+external open class Z__0 : Z<Number>
```

Comment:
Is this entity is required?
Why we can't just use 
`external open class B(a: Number) : Z<Number>`?

Reply:
Seems like, this is the behavior of default parameters transformations. 
This is not, what I've implemented with that changes.